### ISSUE TITLE
Reduce running times by minimizing the number of calls to the Reddit API

### ISSUE THREAD

**Issue by umitkaanusta** [umitkaanusta/reddit-detective#26]
## Problem

Currently, the main reason for high running times is the rate limit of the Reddit API. **So we should minimize the number of calls to reduce running times.**

## Validation of the Problem

**Test code:**
```python3
# Assume api_ is declared and Neo4j is started
net = RedditNetwork(
        driver=driver_,
        components=[
            Comments(Redditor(api_, "Anub_Rekhan", limit=5))
        ]
    )
net.run_cypher_code()
```

The rate-limit evasion function in PRAW is called 59 times by the code above.

The code creates 18 nodes and 22 relationships atm. 59 is simply too much for such a small graph.

**cProfile results:**

![](https://i.imgur.com/kRHy5w6.png)

![](https://i.imgur.com/HOqhO4Y.png)

## Why are there so many requests?

TL;DR: The current version sends an unnecessary amount of requests. 

**Num of calls - Redditor object itself:** **1**

**Num of calls - Getting the complete list of comments:** **5** (2 items in the list of 11 comments are duplicate. That'll be fixed) 

**Num of calls - Merge and Link Comments:** **33**; At each iteration: 3 (1 Comment, 1 Author, 1 Submission). 3 * 11 = 33

**Num of calls - Merge and Link Submissions:** **22**; At each iteration: 2 (1 Subreddit, 1 Author). 2 * 11 = 22 (There shouldn't be 11 submissions instead of 2. That'll be fixed) 

The total number of calls is 61 (Same with the number in the Call Graph)

## What to do?
- "Merge code" generation shouldn't require a `data_models` object. Maybe abstract? (Affects 55 calls in the example above)
- Find other ways of getting the complete list of comments (Affects 5 calls in the example above)

**Comment by umitkaanusta** [comment]
I am finally able to continue working on the update, things are going good i guess:
![rd_perf_update_new_situation](https://user-images.githubusercontent.com/53405015/136016463-6e6d6672-207e-4828-b612-7385fdcacb18.PNG)
